packages/discovery-react-components/src/components/DiscoverySearch/DiscoverySearch.tsx

```diff
@@ -417,26 +417,17 @@ const DiscoverySearch: FC<DiscoverySearchProps> = ({
     fetchFields();
   }, [fetchFields]);
 
-  const api = useMemo((): SearchApiIFC => {
-    return {
-      performSearch: handleSearch,
-      fetchAggregations: handleFetchAggregations,
-      fetchAutocompletions: handleFetchAutocompletions,
-      fetchDocuments: handleFetchDocuments,
-      setSelectedResult: handleSetSelectedResult,
-      setAutocompletionOptions,
-      setSearchParameters,
-      setIsResultsPaginationComponentHidden,
-      fetchFields: handleFetchFields
-    };
-  }, [
-    handleSearch,
-    handleFetchAggregations,
-    handleFetchAutocompletions,
-    handleFetchDocuments,
+  const api = {
+    performSearch: handleSearch,
+    fetchAggregations: handleFetchAggregations,
+    fetchAutocompletions: handleFetchAutocompletions,
+    fetchDocuments: handleFetchDocuments,
+    setSelectedResult: handleSetSelectedResult,
+    setAutocompletionOptions,
     setSearchParameters,
-    handleFetchFields
-  ]);
+    setIsResultsPaginationComponentHidden,
+    fetchFields: handleFetchFields
+  };
```

Comment:
Most of these functions are either already a result of `useCallback` or `useState`, so no need to wrap with `useMemo`.